test/WebJobs.Script.Tests.Integration/CSharpEndToEndTests.cs

```diff
@@ -338,6 +340,248 @@ public CSharpEndToEndTests(TestFixture fixture) : base(fixture)
+
+            var content = await response.Content.ReadAsStringAsync();
+
+            var expectedResponseObject = new HttpResponseMessage();
+            expectedResponseObject.Deserialize(content);
+
+            Assert.NotNull(expectedResponseObject.Headers.GetValues("ResponseServerDateTime"));
+            Assert.Null(expectedResponseObject.Content);
+            Assert.Null(expectedResponseObject.RequestMessage);
+        }
+
+        [Fact]
+        public async Task Proxy_Http_Request_Utf32Content()
+        {
+            var actualRequestObject = new HttpRequestMessage(HttpMethod.Post, "http://httpbin.org/post");
+
+            actualRequestObject.Content = new StringContent("{\"Key\":\"HELLO こんにちは добры дзень سلام 你好 नमस्ते  Здравствуйте\"}", Encoding.UTF32, "application/json");
+
+            var originalRequest = actualRequestObject.Serialize();
+
+            HttpRequestMessage request = new HttpRequestMessage
+            {
+                RequestUri = new Uri(string.Format("http://localhost/api/ProxyHttpRequest")),
+                Method = HttpMethod.Post,
+                Content = new StringContent(originalRequest)
+            };
+            request.SetConfiguration(Fixture.RequestConfiguration);
+
+            Dictionary<string, object> arguments = new Dictionary<string, object>
+            {
+                { "req", request },
+                { ScriptConstants.SystemTriggerParameterName, request }
+            };
+            await Fixture.Host.CallAsync("ProxyHttpRequest", arguments);
+
+            HttpResponseMessage response = (HttpResponseMessage)request.Properties[ScriptConstants.AzureFunctionsHttpResponseKey];
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            var content = await response.Content.ReadAsStringAsync();
+
+            var expectedRequestObject = new HttpRequestMessage();
+            expectedRequestObject.Deserialize(content);
+
+            Assert.Equal(actualRequestObject.RequestUri.AbsoluteUri, expectedRequestObject.RequestUri.AbsoluteUri);
+            Assert.Equal(actualRequestObject.Content.Headers.ContentType.MediaType, expectedRequestObject.Content.Headers.ContentType.MediaType);
+            Assert.Equal(actualRequestObject.Content.Headers.ContentType.CharSet, expectedRequestObject.Content.Headers.ContentType.CharSet);
+
+            var actualContentString = await actualRequestObject.Content.ReadAsStringAsync();
+            var expectedContentString = await expectedRequestObject.Content.ReadAsStringAsync();
+
+            Assert.Equal(actualContentString, expectedContentString);
+        }
+
+        [Fact]
+        public async Task Proxy_Http_Request_BinaryContent()
+        {
+            var actualRequestObject = new HttpRequestMessage(HttpMethod.Post, "http://httpbin.org/post");
+
+            var bytes = File.ReadAllBytes(Path.Combine(Environment.CurrentDirectory, @"TestScripts\CSharp\ProxyHttpRequest\image.png"));
```

Comment:
Can we avoid having an image checked into the repo for this test? E.g. just new up some binary data in this test, e.g. a simple byte array or something?